### PR TITLE
Add travis build target for node.js 0.12 and io.js

### PR DESCRIPTION
**Pull request by bausmeier**
Also cleaned up the prepublish and test scripts.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -60,7 +60,7 @@
     "supertest": "^0.15.0"
   },
   "scripts": {
-    "prepublish": "./node_modules/grunt-cli/bin/grunt build",
-    "test": "./node_modules/grunt-cli/bin/grunt test"
```
Comment (rcrichton) [inline package.json]: Does this automatically run against the local grunt in node_modules? Or does it require grunt to be installed globally?

**Comment by bausmeier** [comment]
@rcrichton We obviously want to support node 0.12, but I'm not sure about io.js. @debonair tested it out (https://github.com/jembi/openhim-core-js/issues/325) and didn't find any issues. Should we run the tests against io.js on travis or should I take it out?

**Comment by rcrichton** [comment]
@bausmeier I think its fine to test against it. It may be something for us to consider running it on. I wonder if we should remove 0.11 as its unstable and rather just focus on 0.12 now.

File: package.json (hunk 1 above)
Comment (bausmeier) [inline package.json]: Yes it adds dependencies to the path. See https://docs.npmjs.com/misc/scripts#path.

Comment (rcrichton) [inline package.json]: Oh cool. I didn't know it did that :)

**Comment by bausmeier** [comment]
I think we should leave 0.11 in at least until we update the installation instructions in the readme.

**Comment by rcrichton** [comment]
Cool, I'm fine with that.

**Pull request merged**